crates/proof-of-sql/src/proof_primitive/dory/dynamic_dory_structure.rs

```diff
@@ -67,23 +57,22 @@ pub(crate) const fn row_and_column_from_index(index: usize) -> (usize, usize) {
 
 /// Returns the index of data where the (row, column) belongs.
 #[allow(dead_code)]
-pub(crate) const fn index_from_row_and_column(row: usize, column: usize) -> usize {
+pub(crate) fn index_from_row_and_column(row: usize, column: usize) -> Option<usize> {
```

Comment:
NIT: Is there a reason you removed `const`?

Reply:
Yes, because of compilation errors it introduced. Adding the return `Option` and keeping `const` on the function produced two errors:

1. `error[E0015]: cannot call non-const operator in constant functions` referring to the `(row, column) != (1, 0)`
2. `error[E0015]: cannot call non-const fn __core::bool::<impl bool>::then_some::<usize> in constant functions` referring to the `.then_some(width_of_row * (row - width_of_row / 2) + column)`